conda_smithy/schema.py

```diff
@@ -0,0 +1,1240 @@
+# This model is also used for generating and automatic documentation for the conda-forge.yml file.
+# For an upstream preview of the documentation, see https://conda-forge.org/docs/maintainer/conda_forge_yml.
+
+import json
+from enum import Enum
+from inspect import cleandoc
+from typing import Any, Dict, List, Literal, Optional, Union
+
+import yaml
+from pydantic import BaseModel, Field, create_model, ConfigDict
+
+from conda.base.constants import KNOWN_SUBDIRS
+
+try:
+    from enum import StrEnum
+except ImportError:
+    from backports.strenum import StrEnum
+
+
+from conda_smithy.validate_schema import (
+    CONDA_FORGE_YAML_DEFAULTS_FILE,
+    CONDA_FORGE_YAML_SCHEMA_FILE,
+)
+
+
+class Nullable(Enum):
+    """Created to avoid issue with schema validation of null values in lists or dicts."""
+
+    null = None
+
+
+#############################################
+######## Choices (Enum/Literals) definitions #########
+#############################################
+
+conda_build_tools = Literal[
+    # will run vanilla conda-build, with system configured / default solver
+    "conda-build",
+    # will run vanilla conda-build, with the classic solver
+    "conda-build+classic",
+    # will run vanilla conda-build, with libmamba solver
+    "conda-build+conda-libmamba-solver",
+    # will run 'conda mambabuild', as provided by boa
+    "mambabuild",
+]
+
+
+class CIservices(StrEnum):
+    azure = "azure"
+    circle = "circle"
+    travis = "travis"
+    appveyor = "appveyor"
+    github_actions = "github_actions"
+    drone = "drone"
+    woodpecker = "woodpecker"
+    default = "default"
+    emulated = "emulated"
+    native = "native"
+    disable = "None"
+
+
+class BotConfigAutoMergeChoice(StrEnum):
+    VERSION = "version"
+    MIGRATION = "migration"
+
+
+class BotConfigSkipRenderChoices(StrEnum):
+    GITIGNORE = ".gitignore"
+    GITATTRIBUTES = ".gitattributes"
+    README = "README.md"
+    LICENSE = "LICENSE.txt"
+    GITHUB_WORKFLOWS = ".github/workflows"
+
+
+class BotConfigInspectionChoice(StrEnum):
+    HINT = "hint"
+    HINT_ALL = "hint-all"
+    HINT_SOURCE = "hint-source"
+    HINT_GRAYSKULL = "hint-grayskull"
+    UPDATE_ALL = "update-all"
+    UPDATE_SOURCE = "update-source"
+    UPDATE_GRAYSKULL = "update-grayskull"
+
+
+class BotConfigVersionUpdatesSourcesChoice(StrEnum):
+    CRAN = "cran"
+    GITHUB = "github"
+    INCREMENT_ALPHA_RAW_URL = "incrementalpharawurl"
+    LIBRARIES_IO = "librariesio"
+    NPM = "npm"
+    NVIDIA = "nvidia"
+    PYPI = "pypi"
+    RAW_URL = "rawurl"
+    ROS_DISTRO = "rosdistro"
+    # CRAN = "CRAN"
+    # GITHUB = "Github"
+    # INCREMENT_ALPHA_RAW_URL = "IncrementAlphaRawURL"
+    # LIBRARIES_IO = "LibrariesIO"
+    # NPM = "NPM"
+    # NVIDIA = "NVIDIA"
+    # PYPI = "PyPI"
+    # RAW_URL = "RawURL"
+    # ROS_DISTRO = "ROSDistro"
```

Comment:
What about these comments? I think currently the bot only supports the lowercase version